### DISCUSSION TITLE
GitHub Pages : stuck with "Certificate Request Error"

### DISCUSSION THREAD

**Discussion by server-KaTys** [community/community#31030] [Other Feature Feedback, Questions, & Ideas] [answered]
Hello,

I have deployed GitHub Pages from the [`server-KaTys/status`](https://github.com/server-KaTys/status) at the [`status.katys.cf`](https://status.katys.cf) custom domain, which used to work, but is stuck with a *Certificate Request Error* for weeks now.

The `dig status.katys.cf +noall +answer -t A` command outputs :
```
status.katys.cf.	3600	IN	CNAME	server-katys.github.io.
server-katys.github.io.	3600	IN	A	185.199.111.153
server-katys.github.io.	3600	IN	A	185.199.108.153
server-katys.github.io.	3600	IN	A	185.199.109.153
server-katys.github.io.	3600	IN	A	185.199.110.153
```
But, Pages settings indicate :
![Screenshot from 2022-08-27 12-39-21](https://user-images.githubusercontent.com/105603274/187026610-6e3c757d-f6d5-4224-b8ea-b9772695e3be.png)

Regarding similar discussions :
- #22052 and #23372 were caused by a missing `CNAME` record, which is present in my case ;
- #23410 was caused by DNSSEC, but I don't have that feature ;
- #22811 was resolved by itself the next day, while I'm experiencing an issue for weeks.

I also deployed GitHub Pages from [`server-KaTys/KaTys`](https://github.com/server-KaTys/KaTys) at [`katys.cf`](https://katys.cf/), which works perfectly.

Thanks.

PS: this is a *machine user* ([GitHub Docs, section *Personal accounts*, § 4](https://docs.github.com/en/get-started/learning-about-github/types-of-github-accounts#personal-accounts)). Further replies will be submitted from @KaKi87.

**Comment by queenofcorgis** [comment, marked as answer]
Moving this into our Pages category for better visibility - @yoannchaudet would you mind taking a look please?

**Reply by KaKi87** [reply]
Hello,

The person you mentioned was already on vacation when you did, and still is today, would it be possible to please ask someone else who would be available ?

Thanks.

**Reply by yoannchaudet** [reply]
Hello there,

I don't see anything wrong with your DNS configuration now. DNS propagation is not an exact science and this could have tripped the initial certificate request. I just kicked the process again and your cert was issued. It will take up to 10 minutes to show up everywhere in our CDN.

Cheers,

**Reply by KaKi87** [reply]
It now works fine, thanks !
Would it be possible to allow end users to trigger that process restart ?
I think many people would gain much time from this.
Thanks

**Reply by yoannchaudet** [reply]
It is actually automatic anytime you access the Pages settings for your repo.

**Reply by KaKi87** [reply]
When opening DevTools, I indeed spotted that request performed every X seconds, but it has never done anything even when letting it run for hours.
Thanks

**Reply by AryanBhaiya** [reply]
I am facing the same problem Can any of you help me?

**Comment by holtzy** [comment]
Hi @yoannchaudet.

I'm facing exactly the same issue as @server-KaTys. I read the same doc/post/discussion and made the same conclusion.

Any chance you could look at my setup and re-trigger the SSL certificate pipeline for me too?
https://github.com/holtzy/The-Python-Graph-Gallery/settings/pages

I'm truly desperate with all my traffic being stopped 😔

Dig for `www.python-graph-gallery.com`
![image](https://github.com/community/community/assets/10254401/c0546ddb-e3fd-4748-a742-b7c85da6c892)

And for `python-graph-gallery.com`
![image](https://github.com/community/community/assets/10254401/9e0aaed3-8eea-4e05-b993-925edce64106)

Thanks!

**Reply by KaKi87** [reply]
Hello, currently your site seems to work fine. ^^

**Reply by holtzy** [reply]
Thanks @KaKi87.
I was so desperate that I moved to Vercel. But I would really like to be back to GitHub pages if I find a way to resolve my issue..

**Comment by ossikoski** [comment]
Hey @yoannchaudet @KaKi87 , I'm having same kind of issues. 

I got it to work already this week, but after making some changes with just the content of the website, I cannot get TLS certificate or not even the DNS check to complete. I have checked DNS settings of my provider, github pages settings etc. and everything should be as before back when it worked. The odd thing, when I click "check again", I get a message saying "Certificate already exists for www.kaivomies.fi in state 'bad_authz'. Resuming the process." I cannot find anything about that.

Any help is appreciated, thank you!

**Reply by yoannchaudet** [reply]
https://www.kaivomies.fi has a cert and works fine.

I highly suggest you also point `kaivomies.fi` at GitHub Pages. This will give you a new certificate but it would be more "web" friendly.

**Comment by ronskoro** [comment]
Hi, I have been trying for several days now to solve the same certificate request issue. Could you also create a new SSL certificate for me  please?
I am in the organization Theis-lab and the project is archmap. I want to configure www.archmap.bio to work with GIthub Pages.

**Reply by natefillups** [reply]
How did you end up getting this fixed?

**Comment by lealceldeiro** [comment]
Same issue happening for me. 2024-04-21 :(

**Reply by LiteBrite82** [reply]
have you tried what worked for these folks? https://github.com/orgs/community/discussions/57680

**Reply by LiteBrite82** [reply]
https://github.com/orgs/community/discussions/37200

**Reply by lealceldeiro** [reply]
> have you tried what worked for these folks? https://github.com/orgs/community/discussions/57680

Hey, @LiteBrite82 ,

Indeed, setting my Cloudflare DNS records to proxy only worked.

Thanks for following up 👍

**Comment by wonder-sk** [comment]
I have started to have this "Certificate Request Error" problem in a repository that previously worked fine for a few months (https://github.com/qgis-sk/uc2024-web), but a couple of days ago the certificate has expired, and was not refreshed since.

DNS check passes fine...

I have also tried to remove the custom domain and add it back (twice), but with the same error:

![image](https://github.com/community/community/assets/193367/cc2ab9cd-282f-4bbf-8eb5-fb7cf600fc66)

(this message stays the same for hours)

Checked with letsdebug.net and things seem fine there: https://letsdebug.net/uc2024.qgis.sk/1899521

I am not sure what else should I try to get the certificates working again...

**Reply by DorianMazur** [reply]
This is broken github feature, I suggest to use something else. At least they could give more detailed error response. It's always generic and not working 90% of the time.

**Reply by LiteBrite82** [reply]
Check out multiple solutions from others; there are so many little things that can cause these errors.
https://github.com/orgs/community/discussions/23050
https://github.com/orgs/community/discussions/37200
https://github.com/orgs/community/discussions/57680

**Reply by LiteBrite82** [reply]
@wonder-sk, @DorianMazur, @lealceldeiro  

DNS and certificate issues have been coming up a ton in the last few weeks. It seems like we have a ton of threads with lots of answers; but its very fragmented. I'm tempted to start and pin a new discussion that captures and summarizes the 10 or so discussions on DNS and TSL over the last 2 years. 

From your perspectives, do you feel that would help or compound the issue (finding helpful answers buried in threads)?

Any thoughts?

**Reply by lealceldeiro** [reply]
> @wonder-sk, @DorianMazur, @lealceldeiro  
> 
> DNS and certificate issues have been coming up a ton in the last few weeks. It seems like we have a ton of threads with lots of answers; but its very fragmented. I'm tempted to start and pin a new discussion that captures and summarizes the 10 or so discussions on DNS and TSL over the last 2 years. 
> 
> From your perspectives, do you feel that would help or compound the issue (finding helpful answers buried in threads)?
> 
> Any thoughts?

I think that would help, yeah.

**Reply by DorianMazur** [reply]
I think it would be helpful to have more details related to the certificate generation error. Because now it's a black box. I'm sure the DNS are set correctly and even github checks them correctly, but for some reason the certificate hasn't been created for 10 days now...

**Reply by LiteBrite82** [reply]
@DorianMazur 
Im assuming you tried these trouble shooting steps: [Securing your GitHub Pages site with HTTPS](https://docs.github.com/en/pages/getting-started-with-github-pages/securing-your-github-pages-site-with-https)

**Reply by yoannchaudet** [reply]
@wonder-sk 

You have an extra `CNAME` that should not be here:

<img width="990" alt="image" src="https://github.com/community/community/assets/14911070/840ed111-6635-428c-b43e-1939055a0057">

You have to pick, either `A` records or a `CNAME`. Not both. This is what's preventing your certificate from being obtained.

**Reply by wonder-sk** [reply]
@yoannchaudet thanks a lot for the hint! I have updated the `uc2024.qgis.sk` domain config so that it uses `A` records instead of `CNAME` ... and it looks like the certificates finally got refreshed :tada:

A couple of confusing bits you may want to address at Github Pages:

- my domain `uc2024.qgis.sk` did not use both `CNAME` and `A` records. It only used one `CNAME` record, as you can see in your output from dig - the `A` entries were for the `qgis.sk` domain (I am using `qgis.sk` domain with github pages from another git repo)

- with the original configuration (using `CNAME`), the setup worked fine for a couple of months (since January 2024), but then the certificate failed to renew, without any change in my DNS configuration

- with the original configuration, I was getting "DNS check successful", making me think there's nothing to worry about my DNS setup. With the new configuration (using `A` records), github tells me there's a problem, yet now the certificate got renewed:

  ![github-pages-final](https://github.com/community/community/assets/193367/a5a78971-488d-471b-a809-9c2c71ab8b4a)


For the record, this is the new DNS config for the domain:

![image](https://github.com/community/community/assets/193367/80665ed4-8422-4e25-8514-bdcdf43499b7)

**Reply by yoannchaudet** [reply]
I misread the output of the command! Sorry for that.

We will relax some validation logic but at the moment, a `CNAME` will only be allowed if it points to `<user or org name>.github.io`. Like we document in https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/managing-a-custom-domain-for-your-github-pages-site#configuring-a-subdomain.

**Comment by Zesaa** [comment]
Thanks

**Comment by brianary** [comment]
Same problem for [brianary/dataelemental](https://github.com/brianary/dataelemental) for four days.

### dig output

```txt
dataelemental.dev.	300	IN	A	185.199.108.153
dataelemental.dev.	300	IN	A	185.199.109.153
dataelemental.dev.	300	IN	A	185.199.111.153
dataelemental.dev.	300	IN	A	185.199.110.153
```

### Custom domain

<img width="1053" height="438" alt="image" src="https://github.com/user-attachments/assets/635b5d5b-7a0e-415e-af2a-653c933a3192" />

**Reply by brianary** [reply]
The above dig output came from https://www.digwebinterface.com/ .

When I run `dig` locally, I don't seem to get a response:

```txt
> dig dataelemental.dev

; <<>> DiG 9.18.39-0ubuntu0.22.04.1-Ubuntu <<>> dataelemental.dev
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: SERVFAIL, id: 11325
;; flags: qr rd ra; QUERY: 1, ANSWER: 0, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 65494
;; QUESTION SECTION:
;dataelemental.dev.		IN	A

;; Query time: 70 msec
;; SERVER: 127.0.0.53#53(127.0.0.53) (UDP)
;; WHEN: Sun Sep 14 10:53:17 PDT 2025
;; MSG SIZE  rcvd: 46

> dig dataelemental.dev +noall +answer -t A
> dig webcoder.info +noall +answer -t A    
webcoder.info.		300	IN	A	185.199.110.153
webcoder.info.		300	IN	A	185.199.108.153
webcoder.info.		300	IN	A	185.199.111.153
webcoder.info.		300	IN	A	185.199.109.153
```

(webcoder.info is another GH Pages site I've had for much longer.)

I'm not sure why dig doesn't return the A records if GH thinks the DNS check was successful.

**Reply by brianary** [reply]
It turns out that name.com doesn't support DNSSEC, which I had at some point enabled for the new domain. Disabling that fixed everything!

**Comment by odanieldamasio** [comment]
Hi, I have been trying for several days now to solve the same certificate request issue. Could you also create a new SSL certificate for me please?
I am in the organization Theis-lab and the project is archmap. I want to configure [www.archmap.bio](http://www.archmap.bio/) to work with GIthub Pages.

**Comment by gmagno** [comment]
Hi @yoannchaudet, I'm having exactly the same issue. May I kindly ask you trigger the process of re-issuing the cert. Since the automatic flow is definitely not working for me :pray:

The repo: https://github.com/gmagno/gmagno.dev

---------

<img width="1067" height="1005" alt="image" src="https://github.com/user-attachments/assets/82390c67-42f9-41ac-8c0e-3dc29989abb4" />

---------

<img width="1396" height="798" alt="image" src="https://github.com/user-attachments/assets/f271c2de-0de0-4506-a6fd-5c4b75093719" />

**Comment by albertoridolfi** [comment]
@yoannchaudet I am seeing the same stuck GitHub Pages certificate pipeline for ridf.net and www.ridf.net. The authenticated Pages DNS health endpoint reports both hosts as valid, served by Pages, HTTPS-eligible, unproxied, with no conflicting/non-GitHub IPs and no CAA error. DNS uses all four documented apex A records, all four AAAA records, and www CNAMEs to albertoridolfi.github.io. The custom domain is verified and Pages builds succeed, but the Pages API certificate remains in state `new` and www still serves the generic *.github.io certificate. No new certificate appears in CT logs. I have already removed/re-added the domain, left it detached before reattaching, rebuilt Pages, and allowed fresh requests to remain uninterrupted beyond the documented provisioning window.

Update: to rule out stale repository-level state, I duplicated the homepage and history into a fresh Pages repository, https://github.com/albertoridolfi/ridf-pages-fallback, successfully built it, detached the domain from the original user site, and attached the verified domain to the fresh project site. A new Pages DNS health run is again completely valid, but the fresh site immediately has the same certificate state `new`. This strongly points to a domain/account-level Pages certificate backend job rather than the repository, build, or DNS. Could a GitHub Pages engineer please inspect or re-kick issuance as was done in this thread?
